### ISSUE TITLE
Stacking with evaluation set

### ISSUE THREAD

**Issue by goolig** [awslabs/autogluon#692]
I'm working with patient data collected for a few years. 
The last year of the data is used as a test set and the year before is kept for validation.

According to the documentation, it is not possible to enable stacking and use a validation set. If stacking is enabled, the validation set is also used for training, I assume the score reported is the out-of-bag score. However, in my case, this badly overfits the data, the score on the test set is low while the out-of-bag score is relatively good, it will be more beneficial in this case to report the score on the validation set and not use it for training, this will enable me to select the best performing model for my needs.

Using hyperparameter tunning instead of stacking and maximizing the score on the validation set gives better performance, which is not recommended in the documentation.

Am I missing something? Perhaps there is a workaround to enable scoring of the validation set?

Thank you all for your great job!

**Comment by Innixma** [comment]
Thanks for the feature request! This is a good feature to have. AutoGluon does not currently support specifying a custom validation set for stacking as we use out-of-fold predictions. However, if you have co-variate shift in your data, this can result in heavy overfitting and worse results then if the validation set was used.

Adding this requires updating the stack ensemble API to take in validation data and early-stop/score using it rather than the out-of-fold predictions. This is do-able, but is a fairly complicated effort. I will see about this making it into the 0.1 release as I think it is a very useful feature.

In the meantime, you may be able to get better results by removing the time feature from your data or doing feature engineering on it so that it is simply dayofweek, monthofyear, etc. Tabular models often perform poorly when values are all smaller in training than in validation/test (such as seconds_from_epoch if data is split by time).

Another option is to use your validation data as input to `predictor.leaderboard(val_data)` in order to find the best scoring model on your validation data.

**Comment by Innixma** [comment]
This has been added in #1105 and will be available in v0.3 release. You can specify `use_bag_holdout=True` in `predictor.fit` to enable during bagging/stacking.
